elliptic-curve-crate/src/weierstrass/public_key.rs

```diff
@@ -101,6 +102,35 @@ where
     }
 }
 
+impl<C: Curve> PublicKey<C>
+where
+    C: FixedBaseScalarMul,
+    <C::ScalarSize as Add>::Output: Add<U1>,
+    CompressedPoint<C>: From<C::Point>,
+    UncompressedPoint<C>: From<C::Point>,
+    CompressedPointSize<C::ScalarSize>: ArrayLength<u8>,
+    UncompressedPointSize<C::ScalarSize>: ArrayLength<u8>,
+{
+    /// Compute the [`PublicKey`] for the provided [`SecretKey`].
+    ///
+    /// The `compress` flag requests point compression.
+    pub fn from_secret_key(secret_key: &SecretKey<C::ScalarSize>, compress: bool) -> Option<Self> {
+        let ct_option = C::mul_base(secret_key.secret_scalar());
+
+        if ct_option.is_some().into() {
+            let affine_point = ct_option.unwrap();
+
+            if compress {
+                Some(PublicKey::Compressed(affine_point.into()))
+            } else {
+                Some(PublicKey::Uncompressed(affine_point.into()))
+            }
+        } else {
+            None
+        }
```

Comment:
I was able to eliminate the `Default` bound on `AffinePoint` by having this return an `Option` rather than a `CtOption` as soon as scalar multiplication is complete (i.e. unwrapping the `CtOption` containing the public key).

Since we're dealing with a public key after scalar multiplication is complete, this seems ok to me.